src/requests/getForecast.js

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable no-console */
+
+import axios from "axios";
+
+const getForecast = (
+  searchText,
+  setSelectedDate,
+  setForecasts,
+  setLocation,
+  setErrorMessage
+) => {
+  let endpoint = "https://mcr-codes-weather-app.herokuapp.com/forecast";
+
+  if (searchText) {
+    endpoint += `?city=${searchText}`;
+  }
+
+  axios
+    .get(endpoint)
+    .then((res) => {
+      setSelectedDate(res.data.forecasts[0].date);
+      setForecasts(res.data.forecasts);
+      setLocation(res.data.location);
+    })
+    .catch((err) => {
+      const { status } = err.response;
+
+      if (status === 404) {
+        setErrorMessage("No such town or city, try again!");
+        console.error("Location is not valid", err);
+      }
+
+      if (status === 500) {
+        setErrorMessage("Oops, servererror, try again later.");
+        console.error("Server error", err);
+      }
+    });
```

Comment:
Would it be a good idea to add a general setError message at the end for any cases that dont match the previous 2?

Reply:
Hmm, not so sure because the documentation says there's only 3 response statuses `200`, `404` and `500` so I think it's unnecessary here, but generally I would definitely agree